smart_open/tests/test_s3.py

```diff
@@ -124,34 +143,41 @@ def test_iter(self):
         put_to_bucket(contents=expected)
 
         # connect to fake s3 and read from the fake key we filled above
+        self.api_calls.clear()
         fin = smart_open.s3.SeekableBufferedInputBase(BUCKET_NAME, KEY_NAME)
         output = [line.rstrip(b'\n') for line in fin]
         self.assertEqual(output, expected.split(b'\n'))
+        self.assertDictEqual(self.api_calls, {'GetObject': 1})
 
     def test_iter_context_manager(self):
         # same thing but using a context manager
         expected = u"hello wořld\nhow are you?".encode('utf8')
         put_to_bucket(contents=expected)
+        self.api_calls.clear()
```

Comment:
Yes, it's there so test-case-specific setup calls don't pollute the count. I don't think it belongs in either `put_to_bucket` or `setUp`, though. Note that it sometimes comes after other test-specific setup calls, e.g. in `test_read_gzip`.

If you prefer, it might be a natural fit for a context manager, e.g.:

```
    def test_iter_context_manager(self):
        # same thing but using a context manager
        expected = u"hello wořld\nhow are you?".encode('utf8')
        put_to_bucket(contents=expected)
        with self.assertApiCalls(GetObject=1):
            with smart_open.s3.SeekableBufferedInputBase(BUCKET_NAME, KEY_NAME) as fin:
                output = [line.rstrip(b'\n') for line in fin]
                self.assertEqual(output, expected.split(b'\n'))
```

That's how it's handled in [django_assert_num_queries](https://pytest-django.readthedocs.io/en/latest/helpers.html#django-assert-num-queries), which is a similar test helper.

Reply:
I refactored to use the `with` approach.